### PR TITLE
Feat/add loki and update to 1.25

### PR DESCRIPTION
**Pull request by nutellinoit**
- [x] Update packages to match 1.25 (`Furyfile.yml` and `kfd.yaml`)
- [x] Add new defaults on `furyctl-defaults.yaml`
- [x] Add patches for nodeselector and tolerations on loki components
- [x] Add patches for loki resources
- [x] Update pomerium policy
- [x] Update JSON schema

### CODE REVIEW

File: templates/distribution/manifests/logging/kustomization.yaml.tpl [hunk 1]
```diff
@@ -3,38 +3,33 @@ apiVersion: kustomize.config.k8s.io/v1beta1
...
+  - {{ print "../" .spec.distribution.common.relativeVendorPath "/modules/logging/katalog/cerebro" }}
+  - {{ print "../" .spec.distribution.common.relativeVendorPath "/modules/logging/katalog/configs" }}
   - {{ print "../" .spec.distribution.common.relativeVendorPath "/modules/logging/katalog/opensearch-dashboards" }}
 {{- if eq .spec.distribution.modules.logging.opensearch.type "single" }}
   - {{ print "../" .spec.distribution.common.relativeVendorPath "/modules/logging/katalog/opensearch-single" }}
 {{- else if eq .spec.distribution.modules.logging.opensearch.type "triple" }}
   - {{ print "../" .spec.distribution.common.relativeVendorPath "/modules/logging/katalog/opensearch-triple" }}
+{{- end }}
+{{- end }}
+{{- if eq .spec.distribution.modules.logging.type "opensearch" }}
```
Comment (ralgozino) [inline templates/distribution/manifests/logging/kustomization.yaml.tpl:21]: I think this one should be loki
```suggestion
{{- if eq .spec.distribution.modules.logging.type "loki" }}
```

File: templates/distribution/manifests/logging/patches/loki.yml.tpl [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+spec:
+  template:
+    spec:
+      containers:
+      - name:
+      {{- if hasKeyAny .spec.distribution.modules.logging.loki "resources" }}
+        resources:
+          {{ .spec.distribution.modules.logging.loki.resources | toYaml | indent 10 | trim }}
+      {{- end }}
+{{- end }}
```
Comment (ralgozino) [inline templates/distribution/manifests/logging/patches/loki.yml.tpl:92]: nitpick: missing new line at EOF

File: templates/distribution/manifests/logging/patches/opensearch.yml.tpl [hunk 3]
```diff
@@ -22,3 +23,4 @@ spec:
       resources:
         requests:
           storage: {{ .spec.distribution.modules.logging.opensearch.storageSize }}
+{{- end }}
```
Comment (ralgozino) [inline templates/distribution/manifests/logging/patches/opensearch.yml.tpl:26]: nitpick: missing new line at EOF

**Review by ralgozino — CHANGES_REQUESTED** [review]

File: templates/distribution/manifests/aws/patches/ebs-csi-driver.yml.tpl [hunk 4]
```diff
@@ -1,7 +1,12 @@
...
+
+---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   annotations:
-    eks.amazonaws.com/role-arn: {{ .spec.distribution.modules.aws.ebsCsiDriver.iamRoleArn }}
+    {{- if $module -}}
+    eks.amazonaws.com/role-arn: {{ $module.ebsCsiDriver.iamRoleArn }}
+    {{- end -}}
```
Comment (g-iannelli) [inline templates/distribution/manifests/aws/patches/ebs-csi-driver.yml.tpl:10]: why add a patch when the only change is within an "if"?

**Review by g-iannelli — CHANGES_REQUESTED** [review]

File: templates/distribution/manifests/aws/patches/cluster-autoscaler.yml.tpl [hunk 5]
```diff
@@ -19,6 +21,8 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   annotations:
-    eks.amazonaws.com/role-arn: {{ .spec.distribution.modules.aws.clusterAutoscaler.iamRoleArn }}
+    {{- if $module -}}
+    eks.amazonaws.com/role-arn: {{ $module.clusterAutoscaler.iamRoleArn }}
+    {{- end -}}
```
[the remark this answers is not in the document]
Comment (omissis) [reply templates/distribution/manifests/aws/patches/cluster-autoscaler.yml.tpl:26]: I added this to guard against accessing a null value, should the aws key be empty.

File: templates/distribution/manifests/aws/patches/ebs-csi-driver.yml.tpl (hunk 4 above)
Comment (omissis) [reply templates/distribution/manifests/aws/patches/ebs-csi-driver.yml.tpl:10]: I added this to guard against accessing a null value, should the aws key be empty.

File: templates/distribution/manifests/aws/patches/load-balancer-controller.yml.tpl [hunk 6]
```diff
@@ -17,6 +19,8 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   annotations:
-    eks.amazonaws.com/role-arn: {{ .spec.distribution.modules.aws.loadBalancerController.iamRoleArn }}
+    {{- if $module -}}
+    eks.amazonaws.com/role-arn: {{ $module.loadBalancerController.iamRoleArn }}
+    {{- end -}}
```
[the remark this answers is not in the document]
Comment (omissis) [reply templates/distribution/manifests/aws/patches/load-balancer-controller.yml.tpl:24]: I added this to guard against accessing a null value, should the aws key be empty.

File: schemas/public/ekscluster-kfd-v1alpha2.json [hunk 7]
```diff
@@ -1159,7 +1166,20 @@
...
+                            "type": {
+                                "const": "opensearch"
+                            }
+                        }
+                    },
+                    "then": {
+                        "required": ["opensearch"]
+                    }
+                }
+            ]
```
Comment (ralgozino) [inline schemas/public/ekscluster-kfd-v1alpha2.json:1182]: should it be a similar block but for `Loki`? If yes, remember to add it to the private schema too.

File: templates/distribution/manifests/logging/patches/loki.yml.tpl [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+kind: StatefulSet
+metadata:
+  name: loki-distributed-ingester
+  namespace: logging
+spec:
+  template:
+    spec:
+      containers:
+      - name: ingester
+        resources:
```
[the remark this answers is not in the document]
Comment (omissis) [reply templates/distribution/manifests/logging/patches/loki.yml.tpl:19]: for they will rely on the package config, or the module, or the common one (in this order). we'll refine those if needed in the future.

**Review by omissis — APPROVED** [review]

**Pull request merged**